config/crd/bases/infrastructure.cluster.x-k8s.io_azuremachinetemplates.yaml

```diff
@@ -190,6 +190,12 @@ spec:
                           id:
                             type: string
                         type: object
+                      identityType:
+                        description: IdentityType is the type of identity used for
```

Comment:
With the identityType, there is different payload requirements for the API ([see the VMIdentity structure](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#virtualmachineidentity)]. 
- System Assigned: just needs the type and then the identity is created by ARM and added to the machine
- User Assigned: the type needs to be specified as well as a list of the Azure resource IDs to the identities to be provisioned on the machine. These identities are supposed to be made ahead of time by the user or some other process (possibly a controller).

Building system assigned and user assigned in one PR might be a bit large. Might be easier to first build out System Assigned, then introduce another PR for User Assigned.

Thoughts, @nader-ziada and @CecileRobertMichon?

Reply:
Agreed. I thought this PR was for System assigned only based on the PR title? 
+1 on removing references to user assigned in this PR and adding them in a later User Assigned dedicated PR.

Reply:
will remove them, just added in the types but wasn't using them